database/user.go

```diff
@@ -1,113 +1,25 @@
 package database
 
 import (
-	"os"
-	"regexp"
-
-	"gitlab.com/NebulousLabs/errors"
-
-	"github.com/NebulousLabs/skynet-accounts/build"
-	"github.com/NebulousLabs/skynet-accounts/lib"
-
-	"gitlab.com/NebulousLabs/fastrand"
 	"go.mongodb.org/mongo-driver/bson/primitive"
-	"golang.org/x/crypto/bcrypt"
 )
 
 // User status tiers.
 const (
-	TierUnconfirmed = iota
+	TierReserved = iota
 	TierFree
-	TierPremium
-)
-
-var (
-	// ErrInvalidEmail is returned when we encounter an invalid email value.
-	ErrInvalidEmail = errors.New("invalid email")
-
-	// emailValidatorRegEx checks if a string is a valid email address.
-	// See https://emailregex.com/
-	emailValidatorRegEx = regexp.MustCompile(`^[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\.[A-Za-z]{2,64}$`)
-
-	// envPepper holds the name of the environment variable for password pepper.
-	// This is not in main in order to avoid a circular dependency.
-	envPepper = "SKYNET_PEPPER"
-
-	// pepper is the system's password pepper.
-	pepper = []byte{}
-
-	// saltSize specifies the length of the Salt []byte
-	saltSize = 32
+	TierPremium5
+	TierPremium20
+	TierPremium80
 )
 
 type (
-	// Email is an email.
-	Email string
-
 	// User represents a Skynet user.
 	User struct {
 		// ID is a hexadecimal string representation of the MongoDB id assigned
 		// to this user object. It is auto-generated by Mongo on insert.
-		ID        primitive.ObjectID `bson:"_id,omitempty" json:"-"`
-		FirstName string             `bson:"firstName" json:"firstName"`
-		LastName  string             `bson:"lastName" json:"lastName"`
-		Email     Email              `bson:"email" json:"email"`
-		Tier      int                `bson:"tier" json:"tier"`
-		Password  []byte             `bson:"password" json:"-"`
-		Salt      []byte             `bson:"salt" json:"-"`
-
-		dep lib.Dependencies
+		ID   primitive.ObjectID `bson:"_id,omitempty" json:"-"`
```

Comment:
why are we hiding it in `json`?

Reply:
Why would the calling service ever need this? We only use JSON to talk to the portal and the portal should never need the MongoID. It's an implementation detail that is not a concern for the caller.

Reply:
yeah but we are also hiding the sub, so basically the user is represented by its tier here, don't we need at least some form of unique id to link to a user outside of the API. 

A stupid example is if a react frontend were to list these users and you'd have to uniquely identify each row.. in react you have to identify children like that.. there a user id of sorts would come in handy.

Also debugging.

Reply:
That makes sense - I added the `sub`.